interfaces/builtin/snapd_control.go

```diff
@@ -35,22 +41,42 @@ const snapdControlBaseDeclarationSlots = `
     deny-auto-connection: true
 `
 
-// http://bazaar.launchpad.net/~ubuntu-security/ubuntu-core-security/trunk/view/head:/data/apparmor/policygroups/ubuntu-core/16.04/snapd-control
 const snapdControlConnectedPlugAppArmor = `
 # Description: Can manage snaps via snapd.
 
 /run/snapd.socket rw,
 `
 
+type snapControlInterface struct {
+	commonInterface
```

Comment:
afaict seems we haven't used this kind of embedding pattern so far, wondering if there's a preference, I suppose it's a question for @zyga

Reply:
I noticed this and I was thinking about it myself. While we haven't done this before I don't think it is a bad practice myself (unless there's something golang-y that I'm not familiar with). Doing it would allow us to simplify a few interfaces that just need to override one method and are happy with the semantics of common interface for others.

Reply:
FWIW it's used in https://github.com/snapcore/snapd/pull/4185 too